Watchman.Discord/Areas/Protection/Services/UserMessagesCountService.cs

```diff
@@ -21,24 +22,45 @@ public UserMessagesCountService(IQueryBus queryBus)
 
         public int CountMessages(Contexts contexts)
         {
-            if (_serverMessagesQuantity == null || _lastUpdated.AddHours(12) < DateTime.UtcNow)
+            if (ShouldReloadCache())
             {
-                _lastUpdated = DateTime.UtcNow;
+                ReloadCache();
+            }
+
+            return GetOneUserFromOneServerCount(contexts.User.Id, contexts.Server.Id);
+        }
 
-                var query = new GetMessagesQuery();
-                var messages = _queryBus.Execute(query).Messages.ToList();
+        private bool ShouldReloadCache()
+        {
+            return _everyServersMessagesQuantity == null || _lastUpdated.AddHours(12) < DateTime.UtcNow;
```

Comment:
czemu akurat 12 godzin?

Reply:
Może być jakakolwiek inna wartość, po prostu chodzi o to, żeby nie odświeżać tego co każdą wiadomość, a wystarczy co jakiś czas. Chodzi o to, że jeśli ktoś już napisał kilkaset wiadomości na serwerze, to nie ma sensu już sprawdzać czy jest spamerem, bo raczej nie będzie. Najbardziej podejrzane są osoby które mają mało wiadomości

Reply:
może ok opcją będzie edytowalna lista w bazie "zaufany użytkownik"
gdzie administratorzy np z punktu strony mogą podejrzeć kto tam jest
dodać tam kogoś lub usunąć 

np są osoby które znasz z przeszłości i wiesz że to dobrzy ludzie

Reply:
Zakładam, że jeśli ktoś już napisał kilkaset wiadomości to jest zaufany, ale mogę wystawić tą listę publicznie i dodać metodę do ServerMessagesCount, aby w przyszłości móc łatwo dodać konfiguracje

Reply:
myśle że to już coś na osobne zadanie